Tests

dankokin opened the pull request.

sergejkoll committed:
fix import

sergejkoll committed:
fix

sergejkoll committed:
fixed tab))

dankokin committed:
added services-tests

adkuz commented on src/services/task_service.go:
```diff
@@ -11,7 +11,7 @@ const (
+type DatastoreTask interface  {
 	GetTaskTId(id int64) (*models.Task)
 	GetTasksAId(id int64) []models.Task
 	GetTasksGId(id int64) []models.Task
```
Что?

adkuz replied:
Нет, я дешифровал. Но я не в криптографию играть пришел.

adkuz commented on src/services/db_service.go:
```diff
@@ -40,3 +43,78 @@ func OpenConfigFile(filename string) (config string){
+		rows.AddRow(
+			expectedUser.Id,
+			expectedUser.Email,
+			expectedUser.Login,
+			expectedUser.Fullname,
+			hash[i],
+			expectedUser.AccVerified)
+
+		mock.ExpectQuery(regexp.QuoteMeta("SELECT * FROM user_table WHERE login = $1")).
+			WithArgs(expectedUser.Login).WillReturnRows(rows)
+	}
+	return &DB{db}, nil
+}
+
+func NewMockDeleteDB(users []models.User, ids []int) (*DB, error) {
+	db, mock, err := sqlmock.New()
+
+	if err != nil {
+		return &DB{}, err
+	}
+	rows := sqlmock.NewRows([]string{"id", "email", "login", "fullname",
+		"password", "acc_verified"})
+	for i := range users {
+		expectedUser := users[i]
+		rows.AddRow(
+			expectedUser.Id,
+			expectedUser.Email,
+			expectedUser.Login,
+			expectedUser.Fullname,
+			expectedUser.Password,
+			expectedUser.AccVerified)
+
+		mock.ExpectQuery(regexp.QuoteMeta("SELECT * FROM user_table WHERE id = $1")).
+			WithArgs(expectedUser.Id).WillReturnRows(rows)
+		mock.ExpectExec(regexp.QuoteMeta("DELETE FROM user_table WHERE id = $1")).
+			WithArgs(ids[i]).
+			WillReturnResult(sqlmock.NewResult(1, 1))
+	}
+	return &DB{db}, nil
+}
```
Зачем вообще этот мок?

adkuz commented on tests/user_test.go:
```diff
@@ -0,0 +1,130 @@
+}
+
+var (
+	expectedUser1 = models.User{
+		Id:	1,
+		Email: "dk@mail.ru",
+		Login: "dk",
+		Fullname: "Dan Kokin",
+		Password: "pass",
+		AccVerified: true,
+	}
+
+	expectedUser2 = models.User{
+		Id:	2,
+		Email: "dk@mail.ru",
+		Login: "dk",
+		Fullname: "Dan Kokin",
+		Password: "pass",
+		AccVerified: true,
+	}
+
+	expectedUser3 = models.User{
+		Id:	42,
+		Email: "dk@mail.ru",
+		Login: "dk",
+		Fullname: "Dan Kokin",
+		Password: "pass",
+		AccVerified: true,
+	}
+)
```
Логины и почты не уникальны. Почему?
По остальным полям вопросов не меньше.

adkuz reviewed (changes requested):
В корне не понял, в чем была цель тестов - мок ни с чем не взаимодействует. В чем тогда его смысл?

dankokin committed:
Added dockerfiles

dankokin committed:
Added dockers

dankokin committed:
[DO NOT MERGE]: start docker-compose

dankokin committed:
database opening error

dankokin committed:
Not working connecting to db

Commit:
change ports

dankokin committed:
Fixed(???) docker

dankokin committed:
Special for Alexandr

Commit:
Improved deployment

- app: remove wait-for-it.sh
- app: add go-dep
- docker-compose: update according to app

dankokin committed:
EST PROBITIE!

Commit:
Fixed dockers

- add services:db launching command

Commit:
Stable deploy as separate containers

Dockerfile:
- remove 'wait' script
- compile server.app inside the container (build.sh)
- add CMD to start server

Dependencies:
- update all (Gopkg.lock and Gopkg.toml)

Deployment (docker-compose.yml):
- remove volumes in app service
- remove cmd in app service

Deleted:
	scripts/start.sh - it is useless (see build.sh)

src/main/main.go
- refactor connection database (try with timeouts)

Commit:
remove main.go from root directory
